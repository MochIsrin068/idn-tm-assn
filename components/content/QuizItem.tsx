import React from "react";
import { Typography, Image } from "antd";
import MainText from "./MainText";

const { Text } = Typography;

type TPropsQuizItem = {
  data: {
    title: string;
    date: string;
    time: string;
    category: string;
    thumbnail: string;
    plays: string;
  };
};

export default function QuizItem({ data }: TPropsQuizItem) {
  return (
    <div className="rounded-lg hover:opacity-90">
      <Image
        src={data.thumbnail}
        alt="thumbnail"
        className="rounded-lg"
        preview={false}
      />

      <MainText isFix secondaryText={data.category} primaryText={data.title} />

      <div className="mt-2 flex items-center">
        <Text className="text-gray-400 font-bold mr-1">{data.plays}</Text>
        <Text className="text-gray-400">Plays</Text>
      </div>
    </div>
  );
}
