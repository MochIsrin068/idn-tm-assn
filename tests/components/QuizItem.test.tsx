import React from "react";
import { render, screen } from "@testing-library/react";
import QuizItem from "../../components/content/QuizItem";

const mockProps = {
  data: {
    title: "Kuis Gambar Ini Bisa Ungkap Apa Ranan Profesimu",
    date: "2022-08-23",
    time: "09.10",
    category: "Education",
    thumbnail:
      "https://sinyalkarir.com/wp-content/uploads/2022/02/Perbedaan-Pekerjaan-dan-Profesi.webp",
    plays: "11.234",
  },
};

describe("tests components QuizItem", () => {
  it("Perform snapshots test QuizItem", () => {
    const inputDefaultTree = render(<QuizItem {...mockProps} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test text category rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.getByText(/Education/i);
    expect(text).toBeInTheDocument();
  });

  it("test text title rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.getByText(
      /Kuis Gambar Ini Bisa Ungkap Apa Ranan Profesimu/i
    );
    expect(text).toBeInTheDocument();
  });

  it("test text plays values rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.getByText(/11.234/i);
    expect(text).toBeInTheDocument();
  });

  it("test text plays title rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.getByText(/Plays/i);
    expect(text).toBeInTheDocument();
  });

  // optional checker and need query by text becouse that not rendered
  it("test text category wrong rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.queryByText(/Business/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text title wrong rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.queryByText(/Kuiss Gaambar/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text plays values wrong rendered on screen", () => {
    render(<QuizItem {...mockProps} />);
    const text = screen.queryByText(/22331.1234/i);
    expect(text).not.toBeInTheDocument();
  });
});
