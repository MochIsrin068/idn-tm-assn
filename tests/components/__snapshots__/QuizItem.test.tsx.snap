// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components QuizItem Perform snapshots test QuizItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-lg hover:opacity-90"
      >
        <div
          class="ant-image"
        >
          <img
            alt="thumbnail"
            class="ant-image-img rounded-lg"
            src="https://sinyalkarir.com/wp-content/uploads/2022/02/Perbedaan-Pekerjaan-dan-Profesi.webp"
          />
        </div>
        <div
          class="mr-4 "
        >
          <span
            class="ant-typography text-gray-400"
          >
            Education
          </span>
          <h5
            class="ant-typography !m-0 line-clamp-3"
          >
            Kuis Gambar Ini Bisa Ungkap Apa Ranan Profesimu
          </h5>
        </div>
        <div
          class="mt-2 flex items-center"
        >
          <span
            class="ant-typography text-gray-400 font-bold mr-1"
          >
            11.234
          </span>
          <span
            class="ant-typography text-gray-400"
          >
            Plays
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-lg hover:opacity-90"
    >
      <div
        class="ant-image"
      >
        <img
          alt="thumbnail"
          class="ant-image-img rounded-lg"
          src="https://sinyalkarir.com/wp-content/uploads/2022/02/Perbedaan-Pekerjaan-dan-Profesi.webp"
        />
      </div>
      <div
        class="mr-4 "
      >
        <span
          class="ant-typography text-gray-400"
        >
          Education
        </span>
        <h5
          class="ant-typography !m-0 line-clamp-3"
        >
          Kuis Gambar Ini Bisa Ungkap Apa Ranan Profesimu
        </h5>
      </div>
      <div
        class="mt-2 flex items-center"
      >
        <span
          class="ant-typography text-gray-400 font-bold mr-1"
        >
          11.234
        </span>
        <span
          class="ant-typography text-gray-400"
        >
          Plays
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
